### ISSUE TITLE
No formatting preferences provided for C/C++ source files

### ISSUE THREAD

**Issue by derwentx** [pkgw/rubbl#185]
providing a preferred formatter configuration file would help contributors ensure their code is formatted in a way that is consistent with the rest of the repo. This is mostly applicable to `casatables/glue.{cc|h}`

One option is a `.clang-format` file: https://clang.llvm.org/docs/ClangFormatStyleOptions.html

Another might be `.editorconfig` : https://editorconfig.org/

**Comment by pkgw** [comment]
You mentioned elsewhere that you were leaning towards clang-format — is that still the case?

Looking at editorconfig, I think it is probably not my preferred option, since I'd want to have a "headless tool" (using their terminology) to run in the CI to check that the formatting has been preserved, and the available options don't seem like great fits for this project.

Something that integrates well with VS Code would be a perk for me personally, since that's what I'm using these days. (After using Emacs for ~20 years I gave up on it last year and frankly have never looked back.)

**Comment by derwentx** [comment]
Yeah, I'm no C++ formatter expert, but clang-format seems to work well with VSCode in my experience.

**Comment by pkgw** [comment]
Cool. I'm very busy this month so I don't plan to be able to work on this any time soon, unfortunately. If you get the chance to file a PR that would be great!
